its/src/test/java/com/sonar/maven/it/suite/JavaTest.java

```diff
@@ -120,6 +122,33 @@ void setJavaVersionProperties() throws IOException {
       entry("sonar.java.target", "1.8"));
   }
 
+  @Test
+  void when_java_home_exists_it_is_used_as_default_sonar_java_jdkHome_value() throws IOException {
+    //verify that property java home exists, it may not be pointing to a jdk
+    Assumptions.assumeTrue(System.getProperties().containsKey("java.home")
+      || System.getenv().containsKey("JAVA_HOME"));
```

Comment:
If we know which of the system property or environment variable is set and we know which has precedence over the other, then we can write the test in a way that helps us detect if we break the precedence rule.

What about checking which one is expected to be used at the start of the test and then check the same source at the end?